From 0fa22a4be5abda44d34cf7c5d0352378c2f9fd54 Mon Sep 17 00:00:00 2001
From: Lars Schmertmann <Lars.Schmertmann@governikus.de>
Date: Mon, 9 Jan 2023 06:54:53 +0100
Subject: Fix warning in q20algorithm.h when xcodebuild is used

q20algorithm.h:150:20: error: unused function template 'operator()'
q20algorithm.h:163:20: error: unused function template 'operator()'
q20algorithm.h:176:20: error: unused function template 'operator()'

Fixes: QTBUG-109874
Change-Id: If5ccbfffd0b6a53f73f221b45033dab7e4775d89
---
 src/corelib/global/q20algorithm.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git x/qtbase/src/corelib/global/q20algorithm.h y/qtbase/src/corelib/global/q20algorithm.h
index f670a5dbeee81420659a20b68b448a7422d939ec..24d801b2cdfd993def8c60a2bf60e738c921d2ea 100644
--- x/qtbase/src/corelib/global/q20algorithm.h
+++ y/qtbase/src/corelib/global/q20algorithm.h
@@ -147,7 +147,7 @@ using std::ranges::none_of;
 [[maybe_unused]] inline constexpr struct { // Niebloid
     template <typename InputIterator, typename Sentinel,
               typename Predicate, typename Projection = q20::identity>
-    constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
+    [[maybe_unused]] constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
     {
         while (first != last) {
             if (std::invoke(pred, std::invoke(proj, *first)))
@@ -160,7 +160,7 @@ using std::ranges::none_of;
 [[maybe_unused]] inline constexpr struct { // Niebloid
     template <typename InputIterator, typename Sentinel,
               typename Predicate, typename Projection = q20::identity>
-    constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
+    [[maybe_unused]] constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
     {
         while (first != last) {
             if (!std::invoke(pred, std::invoke(proj, *first)))
@@ -173,7 +173,7 @@ using std::ranges::none_of;
 [[maybe_unused]] inline constexpr struct { // Niebloid
     template <typename InputIterator, typename Sentinel,
               typename Predicate, typename Projection = q20::identity>
-    constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
+    [[maybe_unused]] constexpr bool operator()(InputIterator first, Sentinel last, Predicate pred, Projection proj = {}) const
     {
         while (first != last) {
             if (std::invoke(pred, std::invoke(proj, *first)))
